doc/tutorial/c_tutorial/c_tutorial.rst

```diff
@@ -0,0 +1,148 @@
+
+Tutorial #1 - Reachibility Monitor
+==================================
+
+In our first tutorial we are going to build a program that will store the reachibility information in the network of four nodes: Menlo Park, Santa Barbara, Los Angeles, and Salt Lake City. User will update link status between nodes through CLI and our program will calculate the rest. The purpose of this program is to provide a simple interface to a network operator and return information regarding which node can reach which. Thus, the only concern of network operator is to update link status without having to worry about doing those calculations themself. While, this task can be relatively easy done in our example of four nodes, in a more realistic scenario, the number of nodes can be in the order of hundreds and link status updates can be streamed into the monitoring application.
+
+Let's start with defining our program in the DDlog language. In our program we need to supply source node, destination node, and a link status between them. Now we are ready to define our input relation :code:`Links`. Let's create a file called :code:`t1_reachibility_monitor.dl` and add there a following line representing input relation for our reachibility monitor:
+
+.. code-block::
+    
+	input relation Links(src: string, dst: string, link_status: bool)
+
+However, for the reachibility monitor to be useful it also should provide some output and not just ingest the data. We want our program to output a list of nodes that are currently reachable based on the information in the relation :code:`Links`. Thus, let's add an output relation :code:`ConnectedNodes` and provide a rule to calculate records in this relation. To do that we are going to add few more lines to the :code:`t1_reachibility_monitor.dl` so it will looks as follows:
+
+.. code-block::
+
+	input relation Links(src: string, dst: string, link_status: bool)
+	output relation ConnectedNodes(str: string, dst: string)
+
+	ConnectedNodes(src, dst) :- Links(src, dst, true).
+	ConnectedNodes(src, dst) :- ConnectedNodes(src, intermediate_node),
+				Links(intermediate_node, dst, true).
+
+Let's briefly discuss the rule to calculate :code:`ConnectedNodes` relation. First, we say that any directly connected nodes such that the :code:`link_status` between them is :code:`true` belong to :code:`ConnectedNodes` relation. Furthermore, any two nodes are considered connected if there is path from the source node to the destination node such that all directly connected nodes along the path are also connected. Take a minute here to realize how easy is it to write such rule in DDlog. Isn't is amazing?
+
+With that done, let's compile and start out program.
+
+.. code-block::
+
+	$ ddlog -i t1_reachibility_monitor.dl
+	$ cd t1_reachibility_monitor_ddlog/ && cargo build --release

```

Comment:
maybe you can say that the _ddlog directory is generated by the compiler

Reply:
Do you want to say something about the fact that this will take a while the first time, but be much faster later?